nomad/server.go

```diff
@@ -612,6 +615,18 @@ func (s *Server) State() *state.StateStore {
 	return s.fsm.State()
 }
 
+// Regions returns the known regions in the cluster.
+func (s *Server) Regions() []string {
+	s.peerLock.RLock()
+	defer s.peerLock.RUnlock()
+
+	regions := make([]string, 0, len(s.peers))
```

Comment:
The peers map key is region ID, which would prevent having any duplicates. There might be multiple servers but they are still all nested under the same map key.